### PR TITLE
Update docs to surface CLI filtering syntax

### PR DESCRIPTION
**Pull request by strickvl**
Following a user suggestion in Slack, I updated the docs to offer more pointers to users on how / when to use our filtering syntax.

Also fixed some formatting issues in the CLI docs.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Review by fa9r — APPROVED** [review]
Great idea, thanks for documenting that 🙏

**Comment by strickvl** [comment]
@fa9r added a section on using the `Client` methods as well.

File: docs/book/starter-guide/pipelines/fetching-pipelines.md [hunk 1]
```diff
@@ -155,10 +155,33 @@ zenml pipeline runs list --created "gt:2021-01-01 00:00:00"
 ```
 
 Please [refer to our
-documentation](https://apidocs.zenml.io/0.38.0/cli/#zenml.cli--filtering-cli-output-when-listing)
+documentation](https://apidocs.zenml.io/latest/cli/#zenml.cli--filtering-cli-output-when-listing)
 for full details on the available flags.
 </details>
 
+## Querying the Client
```
Comment (fa9r) [inline docs/book/starter-guide/pipelines/fetching-pipelines.md:162]: Should be h3 since runs is h2

```suggestion
### Querying the Client
```

Comment (fa9r) [reply docs/book/starter-guide/pipelines/fetching-pipelines.md:162]: Actually now that I look at the page, how about we keep it a h2 header but move it to the bottom of the page after "Code Example"?

File: docs/book/starter-guide/pipelines/fetching-pipelines.md [hunk 2]
```diff
@@ -155,10 +155,33 @@ zenml pipeline runs list --created "gt:2021-01-01 00:00:00"
...
+
+You can also access ZenML artifacts and objects directly from the ZenML client,
```
Comment (fa9r) [inline docs/book/starter-guide/pipelines/fetching-pipelines.md:164]: artifacts is mentioned twice

```suggestion
Alternatively, you can also access ZenML objects directly from the ZenML client,
```

**Review by fa9r — CHANGES_REQUESTED** [review]
I'd suggest to move the new section down towards the end of the page since it is not really run-specific (as you wrote, you can fetch all kinds of objects that way)

**Review by fa9r — APPROVED** [review]
🚀

**Pull request merged**